C-Sharp/graphs/TopologicalSorting.cs

```diff
@@ -0,0 +1,130 @@
+        //Adjacency list
+        List<int>[] adjList;
+        public Graph(int v)
+        {
+            vertices = v;
+            adjList = new List<int>[v];
+            for (int i = 0; i < v; i++)
+            {
+                adjList[i] = new List<int>();
+            }
+        }
+        //Add direct edge from vertex u to vertex v
+        public void AddEdge(int u, int v)
+        {
+            adjList[u].Add(v);
+        }
+        //Modified DFS algorithm to traverse the graph
+        public void ModifiedDFS(int v, bool[] visited, Stack<int> stack)
+        {
+            visited[v] = true;
+            List<int> connected_vertices = adjList[v];
+            foreach (var vertex in connected_vertices)
+            {
+                //if the vertex not visited recurse
+                if (!visited[vertex])
+                {
+                    ModifiedDFS(vertex, visited, stack);
+                }
+            }
+            //add the vertex to stack after it is visited and its children
+            stack.Push(v);
+        }
+        //Call Modified DFS and perform topological sorting
+        public void TopologicalSort()
+        {
+            bool[] visited = new bool[vertices];
+            Stack<int> stack = new Stack<int>();
+            for (int i = 0; i < vertices; i++)
+            {
+                if (!visited[i])
+                    ModifiedDFS(i, visited, stack);
+            }
+            while(stack.Count != 0)
+                Console.Write(stack.Pop() + " ");
+        }
+        public void TopologicalSort(int start_vertex)
+        {
+            bool[] visited = new bool[vertices];
+            Stack<int> stack = new Stack<int>();
+            ModifiedDFS(start_vertex, visited, stack);
+            while (stack.Count != 0)
+                Console.Write(stack.Pop() + " ");
+        }
+        static void Main(string[] args)
+        {
+            Console.Write("Enter Number of vertices: ");
+            int n = Convert.ToInt32(Console.ReadLine());
+            Graph g = new Graph(n);
+            Console.Write("Enter number of Edges: ");
+            int num_edges = Convert.ToInt32(Console.ReadLine());
+            while (num_edges != 0)
+            {
+                Console.Write("Enter Number of first Vertex: ");
+                int u = Convert.ToInt32(Console.ReadLine());
+                Console.Write("Enter Number of second Vertex: ");
+                int v = Convert.ToInt32(Console.ReadLine());
+                g.AddEdge(u, v);
+                num_edges--;
+            }
+            Console.WriteLine("Do you want Topological sort of the whole graph (YES/NO): ");
+            string choice = Console.ReadLine();
+            if (choice == "YES")
+            {
+                Console.WriteLine("Topological sort of the graph");
+                g.TopologicalSort();
+            }
+            else
+            {
+                Console.Write("Enter your sratr vertex: ");
```

Comment:
What is `sratr vertex`? I guess its a typo fix it.

Reply:
Done